model/textparse/nhcbparse_test.go

```diff
@@ -0,0 +1,515 @@
+		}, {
+			m:    `ggh_count`,
+			v:    1,
+			lset: labels.FromStrings("__name__", "ggh_count"),
+			es:   []exemplar.Exemplar{{Labels: labels.FromStrings("id", "gaugehistogram-count-test", "xx", "yy"), Value: 4, HasTs: true, Ts: 123123}},
+		}, {
+			m:   "smr_seconds",
+			typ: model.MetricTypeSummary,
+		}, {
+			m:    `smr_seconds_count`,
+			v:    2,
+			lset: labels.FromStrings("__name__", "smr_seconds_count"),
+			es:   []exemplar.Exemplar{{Labels: labels.FromStrings("id", "summary-count-test"), Value: 1, HasTs: true, Ts: 123321}},
+		}, {
+			m:    `smr_seconds_sum`,
+			v:    42,
+			lset: labels.FromStrings("__name__", "smr_seconds_sum"),
+			es:   []exemplar.Exemplar{{Labels: labels.FromStrings("id", "summary-sum-test"), Value: 1, HasTs: true, Ts: 123321}},
+		}, {
+			m:   "ii",
+			typ: model.MetricTypeInfo,
+		}, {
+			m:    `ii{foo="bar"}`,
+			v:    1,
+			lset: labels.FromStrings("__name__", "ii", "foo", "bar"),
+		}, {
+			m:   "ss",
+			typ: model.MetricTypeStateset,
+		}, {
+			m:    `ss{ss="foo"}`,
+			v:    1,
+			lset: labels.FromStrings("__name__", "ss", "ss", "foo"),
+		}, {
+			m:    `ss{ss="bar"}`,
+			v:    0,
+			lset: labels.FromStrings("__name__", "ss", "ss", "bar"),
+		}, {
+			m:    `ss{A="a"}`,
+			v:    0,
+			lset: labels.FromStrings("A", "a", "__name__", "ss"),
+		}, {
+			m:   "un",
+			typ: model.MetricTypeUnknown,
+		}, {
+			m:    "_metric_starting_with_underscore",
+			v:    1,
+			lset: labels.FromStrings("__name__", "_metric_starting_with_underscore"),
+		}, {
+			m:    "testmetric{_label_starting_with_underscore=\"foo\"}",
+			v:    1,
+			lset: labels.FromStrings("__name__", "testmetric", "_label_starting_with_underscore", "foo"),
+		}, {
+			m:    "testmetric{label=\"\\\"bar\\\"\"}",
+			v:    1,
+			lset: labels.FromStrings("__name__", "testmetric", "label", `"bar"`),
+		}, {
+			m:    "foo",
+			help: "Counter with and without labels to certify CT is parsed for both cases",
+		}, {
+			m:   "foo",
+			typ: model.MetricTypeCounter,
+		}, {
+			m:    "foo_total",
+			v:    17,
+			lset: labels.FromStrings("__name__", "foo_total"),
+			t:    int64p(1520879607789),
+			es:   []exemplar.Exemplar{{Labels: labels.FromStrings("id", "counter-test"), Value: 5}},
+			// TODO(krajorama): ct: int64p(1520872607123),
+		}, {
+			m:    `foo_total{a="b"}`,
+			v:    17.0,
+			lset: labels.FromStrings("__name__", "foo_total", "a", "b"),
+			t:    int64p(1520879607789),
+			es:   []exemplar.Exemplar{{Labels: labels.FromStrings("id", "counter-test"), Value: 5}},
+			// TODO(krajorama): ct: int64p(1520872607123),
+		}, {
+			m:    "bar",
+			help: "Summary with CT at the end, making sure we find CT even if it's multiple lines a far",
+		}, {
+			m:   "bar",
+			typ: model.MetricTypeSummary,
+		}, {
+			m:    "bar_count",
+			v:    17.0,
+			lset: labels.FromStrings("__name__", "bar_count"),
+			// TODO(krajorama): ct:   int64p(1520872608124),
```

Comment:
I noticed created timestamp returning nil, whats blocking having this now? I know there are things in progress but lack the context. Maybe we could add a TODO as well to the method.

Reply:
Right, need to add it to the method as well. We had a discussion about CT on CNCF slack. Currently the trivial solution to just simply call the embedded Parser's CreatedTimestamp function doesn't work because it looses exemplars (#15137) and has too much overhead. Waiting for more optimizations to land for CT and then iterate further.

Reply:
LGTM for todo for now